### PR TITLE
Added smoke_tcp_ok

### PR DESCRIPTION
**Pull request by lpicanco**
Added smoke_tcp_ok function, to test a connection using tcp.

### CODE REVIEW

File: smoke.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+t#!/bin/bash
```
Comment (asm89) [inline smoke.sh]: Oops. Can you revert this? :)

**Comment by asm89** [comment]
Can you revert the `t` and squash your commits. I like the addition though! 👍

**Comment by lpicanco** [comment]
Reverted!

**Comment by asm89** [comment]
@lpicanco You need to rebase onto master as some other features were merged recently and it now has conflicts.

**Comment by lpicanco** [comment]
Rebased!

**Pull request merged**